### DISCUSSION TITLE
Locale/language-sensitive case mapping (Turkish case folding)

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#11688] [Ideas]
If you care a whit about internationalization (i18n), force your code to run under the Turkish locale as soon as reasonably possible.

Please add localized case folding support like golang:

GO case folding
https://golang.org/pkg/strings/#ToLowerSpecial
https://golang.org/pkg/strings/#ToUpperSpecial

I can suggest test code below for Turkish.

```v
fn test_turkish_case() {
	lower := 'abcçdefgğhıijklmnoöprsştuüvyz'
	upper := 'ABCÇDEFGĞHIİJKLMNOÖPRSŞTUÜVYZ'
	for i, c in lower {
		assert c == upper[i]
	}
}
```

Vlang has a dreamy syntax and features. I have no doubt that the Turks will be very excited when they discover this language. You don't want to deprive them of this feature :)

**Comment by medvednikov** [comment]
You can be sure that all languages will be supported, just like in Go :)

**Comment by noraj** [comment]
Related: https://github.com/rust-lang/rust/issues/72966
